flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateTest.java

```diff
@@ -415,16 +395,19 @@ public void testRequestBuffersWithRemoteInputChannel() throws Exception {
 		final NetworkEnvironment network = createNetworkEnvironment();
 
 		try {
-			final ResultPartitionID resultPartitionId = new ResultPartitionID();
-			final ConnectionID connectionId = new ConnectionID(new InetSocketAddress("localhost", 5000), 0);
-			addRemoteInputChannel(network, inputGate, connectionId, resultPartitionId, 0);
+			RemoteInputChannel remoteFromUnknown =
+				InputChannelBuilder.newBuilder()
+					.setupFromNetworkEnvironment(network)
+					.buildUnknown(inputGate)
+					.toRemoteInputChannel(InputChannelBuilder.STUB_CONNECTION_ID);
+			IntermediateResultPartitionID resultPartitionId = remoteFromUnknown.getPartitionId().getPartitionId();
+			inputGate.setInputChannel(resultPartitionId, remoteFromUnknown);
 
 			network.setupInputGate(inputGate);
 
 			NetworkBufferPool bufferPool = network.getNetworkBufferPool();
 			if (enableCreditBasedFlowControl) {
-				RemoteInputChannel remote = (RemoteInputChannel) inputGate.getInputChannels()
-					.get(resultPartitionId.getPartitionId());
+				RemoteInputChannel remote = (RemoteInputChannel) inputGate.getInputChannels().get(resultPartitionId);
```

Comment:
Use the previous `remoteFromUnknown` directly, so `IntermediateResultPartitionID resultPartitionId = remoteFromUnknown.getPartitionId().getPartitionId()` could also be removed.